src/rgw/rgw_asio_frontend.cc

```diff
@@ -350,6 +351,23 @@ int AsioFrontend::init()
 
     ldout(ctx(), 4) << "frontend listening on " << l.endpoint << dendl;
   }
+  uid_t uid = ctx()->get_set_uid();
+  gid_t gid = ctx()->get_set_gid();
+  std::string uid_string = ctx()->get_set_uid_string();
+  std::string gid_string = ctx()->get_set_gid_string();
+  if (gid && setgid(gid) != 0) {
+    cerr << "unable to setgid " << gid << ": " << cpp_strerror(errno)
+         << std::endl;
+    return -errno;
+  }
+  if (uid && setuid(uid) != 0) {
+    cerr << "unable to setuid " << uid << ": " << cpp_strerror(errno)
+         << std::endl;
+    return -errno;
+  }
+  if (uid && gid) {
+      ldout(ctx(), 1) << "set uid:gid to " << uid << ":" << gid << " (" << uid_string << ":" << gid_string << ")" << dendl;
+  }
```

Comment:
since this code is essentially duplicated from `global_init()`, can we find a way to move both into a common function? maybe something like `int CephContext::drop_privileges(std::ostream&)`?